### PR TITLE
fix(Core/Spell): implement taunt DR CREATURE_FLAG_EXTRA_OBEYS_TAUNT_DIMINISHING_RETURNS 

### PR DESCRIPTION
**Pull request by sogladev**
## Changes Proposed:

This PR proposes changes to:
-  [x] Core (units, players, creatures, game systems).
-  [ ] Scripts (bosses, spell scripts, creature scripts).
-  [x] Database (SAI, creatures, etc).

CREATURE_FLAG_EXTRA_OBEYS_TAUNT_DIMINISHING_RETURNS flag (extra_flags) is not set on any creature

When set, DR handling is incorrect. Most taunts have an EFFECT_APPLY_AURA  with SPELL_AURA_MOD_TAUNT as an effect AND a SPELL_EFFECT_ATTACK_ME. Currently, the former is handled by DR but SPELL_EFFECT_ATTACK_ME is still handled. Resulting in no taunt DR when the flag is set.

Some special cases have a 3rd effect:
- death grip: grip mechanic
- mocking blow: damage

sidenote: hand of reckoning only has 2 effects, damage is handled as a special case in taunteffect

changes:
- add CREATURE_FLAG_EXTRA_OBEYS_TAUNT_DIMINISHING_RETURNS to some creature_templates
- skip handling of other effects if taunt DR's

I believe TC picked these creatures like so:
```
-- Bosses should have DR taunt immunity
-- Check creatures with CREATURE_FLAG_EXTRA_INSTANCE_BIND and Rank 3 (boss) seems ok or is better all Rank3 creatures??
-- Skip bosses with CREATURE_FLAG_EXTRA_NO_TAUNT
UPDATE `creature_template` SET `flags_extra`=`flags_extra`|0x00080000 WHERE `rank`=3 AND `flags_extra`=`flags_extra`|0x00000001 AND `flags_extra`!=`flags_extra`|0x00000100; 
```

full list of 294 creature_template with names/lvl when selecting from acore_world.creature_template 

https://gist.github.com/sogladev/2bc968578e2ee8976e73573e7105d3a8

## Issues Addressed:

- Closes 

## SOURCE:

The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [ ] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [x] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

TC 335 https://github.com/TrinityCore/TrinityCore/commit/fe38b78c877fd30cc677913229d23bd6f5c15f0b

Quoted from TC 335 issue

> https://wowwiki.fandom.com/wiki/Diminishing_returns
> https://www.mmo-champion.com/threads/640994-Taunt-Diminishing-Returns
> 

Patch 3.3.0 (08-Dec-2009): Taunt Diminishing Returns: We've revised the system for diminishing returns on Taunt so
that creatures do not become immune to Taunt until after 5 Taunts have landed. The duration of the Taunt effect will be  reduced by 35% instead of 50% for each taunt landed. In addition, most creatures in the world will not be affected by 
 Taunt diminishing returns at all. Creatures will only have Taunt diminishing returns if they have been specifically flagged for that behavior based on the design of a given encounter.

## Tests Performed:

This PR has been:
- [x] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [ ] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

requires 2 characters

1. spawn an npc with flag set like patchwerk
2. with character 1: taunt a few times with .cheat cooldown to DR taunt. observe aura DR's
3. with character 2: press taunt. npc should not be taunted

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by avarishd** [comment]
Please tag the original author in a commit.

**Comment by Kitzunu** [comment]
When merging add this unless the author has already added the original author in a commit:

> Co-authored-by: offl <offl@users.noreply.github.com>

**Comment by heyitsbench** [comment]
Does look like there are a bunch of creatures in this query that do not exist in AC's database.

**Comment by Kitzunu** [comment]
Status

**Comment by sogladev** [comment]
Running select on creature ids on TDB 335 and acore give same number of results

added co-author in 86c304e6396fd9aad53ed298d15240bccfa66b89

can be tested

File: data/sql/updates/pending_db_world/rev_1718974840446011698.sql [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+39788, -- Anraphet
+39863, -- Halion
+40142, -- Halion
+40177, -- Forgemaster Throngus
+40484, -- Erudax
+40960, -- Bug 193702
+41270, -- Onyxia
+41376, -- Nefarian
+41378, -- Maloriak
+41442, -- Atramedes
```
Comment (heyitsbench) [inline data/sql/updates/pending_db_world/rev_1718974840446011698.sql:107]: I think every creature below this ID (41442) does not exist in AC's `creature_template` table.

Comment (sogladev) [reply data/sql/updates/pending_db_world/rev_1718974840446011698.sql:107]: 41270 does not exist in acore; neither in TDB. Maybe they did at some point, need to check the list again to remove non-existing entries

**Comment by sogladev** [comment]
Following linked TC335 method, exported NPCs that match 
```
`rank`=3 AND `flags_extra`=`flags_extra`|0x00000001 AND `flags_extra`!=`flags_extra`|0x00000100
```

https://gist.github.com/sogladev/c9d3f09dd868417d5e04fb8d10e97357

**Comment by Nyeriah** [comment]
Gotta remove all of the Cata+ creatures

**Review by Nyeriah — APPROVED** [review]
